torchdata/nodes/samplers/multi_node_weighted_sampler.py

```diff
@@ -0,0 +1,216 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import copy
+from typing import Any, Dict, Mapping, Optional
+
+import torch
+from torchdata.nodes.base_node import BaseNode, T
+from torchdata.nodes.samplers.utils import StopCriteria
+
+from .utils import _get_rank_seed, get_rank_and_world_size
+
+
+class MultiNodeWeightedSampler(BaseNode[T]):
+    """A node that samples from multiple datasets with weights."""
```

Comment:
This class is not super self-explanatory. We should include a lengthy docstring here with examples about the expected behaviour of different settings. See eg MDL's doc + param strings. 

Also can follow up on this after PR but before release, let's make sure the docstring styles / param descriptions / param names etc are all consistent